activerecord/lib/active_record/result.rb

```diff
@@ -34,13 +34,13 @@ module ActiveRecord
   class Result
     include Enumerable
 
-    attr_reader :columns, :rows, :column_types
+    attr_reader :columns, :rows, :types
 
-    def initialize(columns, rows, column_types = {})
+    def initialize(columns, rows, types = [])
       @columns      = columns
       @rows         = rows
       @hash_rows    = nil
-      @column_types = column_types
+      @types        = types
```

Comment:
Can change this into something like

```ruby
@column_types = if column_types.present?
  Hash[@columns.zip(column_types)]
else
  {}
end
```

Then we won't need to make `column_types` a standalone method.

Reply:
We don't want to _store_ type data like this, though, because — and this is the bug — if one name occurs more than once in the `@columns` array, then when we index them in a hash, only one type will be associated with that name.

##### Example:
```ruby
Hash[%w[name name title].zip([Integer, Date, String])] #=> { "name" => Date, "title" => String }
```

Reply:
(Sorry — to complete my thought: ... so we _have_ to store the Array and the Hash can be derived from the Array.)

Reply:
I did end up making this change!